lib/usecase/logout_use_case.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:async';
+
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/repository/oauth_repository.dart';
+import 'package:flutter_survey/local/shared_preference_helper.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:injectable/injectable.dart';
+
+@Injectable()
+class LogoutUseCase extends NoParamsUseCase<void> {
+  final OAuthRepository _repository;
+  final SharedPreferencesHelper _sharedPreferencesHelper;
+
+  const LogoutUseCase(this._repository, this._sharedPreferencesHelper);
+
+  @override
+  Future<Result<void>> call() async {
+    final token = await _sharedPreferencesHelper.getAccessToken();
+    return _repository
+        .logout(token: token)
+        .then((value) => _persistTokenData())
+        .onError<NetworkExceptions>(
+            (err, stackTrace) => Failed(UseCaseException(err)));
+  }
+
+  Result<dynamic> _persistTokenData() {
```

Comment:
I think we should clear the token regardless of the API result in case of logout. Besides, the function name should be `clearToken` instead?

Reply:
@manh-t oh, agree, I forget this point 😂

Reply:
@manh-t Fixed by https://github.com/luongvo/flutter-survey/pull/73/commits/19d5e3da53db88dcf08856debc8a2a976405bc2f.